Remove always-`nil` `setConfigDefaults` error return value in generated clients

tklauser opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

The error returned by `setConfigDefaults` called in
`NewForConfig{,AndClient}` in generated clients is always `nil`. Hence, the error return can be omitted and the respective error checking code can be avoided.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

Jefftree commented:
/ok-to-test
/cc @aojea 
/triage accepted

tklauser commented:
/test pull-kubernetes-unit

dgrisonnet commented:
/remove-sig instrumentation

Jefftree commented:
@tklauser: For easier review, can you separate the func signature/generator changes and the automatic codegen changes into a separate commit?

/lgtm
/cc @sttts

aojea commented:
/test

aojea commented:
/test pull-kubernetes-apidiff-client-go

sttts commented:
> is not too much code churn for not much win?

This.

But looks innocent otherwise.

/approve

tklauser commented:
> is not too much code churn for not much win?

The always-`nil` return value in the generated code is flagged by the [`unparam`](https://github.com/mvdan/unparam) linter which some external users of the client-gen code generator might be using (I found this while trying to enable said linter in github.com/cilium/cilium), so I think this could still have some benefit.

tklauser committed:
Remove always-nil setConfigDefaults error return value in generated clients

The error returned by setConfigDefaults called in
NewForConfig{,AndClient} in generated clients is always nil. Hence, the
error return can be omitted and the respective error checking code can
be avoided.

tklauser committed:
Update generated clients after removing always-nil setConfigDefaults error return value

tklauser commented:
> @tklauser: For easier review, can you separate the func signature/generator changes and the automatic codegen changes into a separate commit?

Done in latest force-push.

tklauser commented:
/test

tklauser commented:
/test all

sttts commented:
/lgtm
/approve

tklauser commented:
/assign @dashpole

(pinging for approval according to https://github.com/kubernetes/kubernetes/pull/129139#issuecomment-2544884562)

dashpole commented:
/approve
